cmd/root.go

```diff
@@ -60,6 +85,7 @@ func processBrokerOpts(cmd *cobra.Command, args []string) error {
 func init() {
 	rootCmd.Flags().SortFlags = false
 
+	rootCmd.Flags().StringVar(&namespace, "namespace", "default", "Watch for objects in desired namespace")
```

Comment:
Could default to `/var/run/secrets/kubernetes.io/serviceaccount/namespace` if it exists maybe? Being explicit is good though.

Reply:
This namespace value is [set on the manager](https://github.com/kubernetes-sigs/controller-runtime/blob/add0b6444f5091f56824b070d28f930090d8deab/pkg/manager/manager.go#L246) and propagated into the cache and informer objects. I looked to see if it could be modified post-creation but that looks to be a convoluted if not impossible operation.

This means we'd need to perform the file check prior to creating the manager object, which is simple enough, but also means in-cluster and out-of-cluster behavior will diverge somewhat. What do you think?

Reply:
I think it's ok as-is, just wanted to bring it up in case there was something straightforward in kubebuilder already.